src/PhpDoc/TypeNodeResolver.php

```diff
@@ -287,10 +290,16 @@ private function resolveIdentifierTypeNode(IdentifierTypeNode $typeNode, NameSco
 				]);
 
 			case 'non-empty-string':
+				return new IntersectionType([
+					new StringType(),
+					new AccessoryNonEmptyStringType(),
+				]);
+
 			case 'non-empty-lowercase-string':
```

Comment:
do we have a test somewhere for this type?

Reply:
Added in https://github.com/phpstan/phpstan-src/pull/3438/commits/9636e83175427d8050c98e23d3a6807118457951